modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/controller/LoginController.scala

```diff
@@ -124,13 +122,20 @@ class LoginController extends CasClientConfigurerAdapter {
     */
   @RequestMapping(value = Array("/local"), method = Array(RequestMethod.POST))
   def userLocalLogin(request: HttpServletRequest, response: HttpServletResponse, @RequestBody jsonNode: JsonNode): Unit = {
-    val login = for {
+    val credentials = for {
       username <- jsonNode.retrive("username").asText()
       password <- jsonNode.retrive("password").asText()
-      user <- userService.find(username, password)
-    } yield user
+    } yield (username, password)
 
-    login match {
+    val user = credentials.flatMap(creds =>
+        userService.find(creds._1, creds._2).orElse(for {
+            ldapLogin <- ldapService.login(creds._1, creds._2)
+            ldapUser <- loadUserFromLdap(ldapLogin.getAttribute("uid").getStringValue)
+              .map(user => userService.find(user.username).getOrElse(userService.create(user, null)))
+          } yield ldapUser)
+    )
```

Comment:
I would not put the ldap login in the route `.../api/v1/login/local` but make a new route `.../api/v1/login/` which tries several logins.